code/lib/builder-manager/src/utils/files.test.ts

```diff
@@ -1,19 +1,44 @@
+import { platform } from 'os';
 import { sanitizePath } from './files';
 
-test('sanitizePath', () => {
-  const addonsDir = '/Users/username/Projects/projectname/storybook';
-  const text = 'demo text';
-  const file = {
-    path: '/Users/username/Projects/projectname/storybook/node_modules/@storybook/addon-x+y/dist/manager.mjs',
-    contents: Uint8Array.from(Array.from(text).map((letter) => letter.charCodeAt(0))),
-    text,
-  };
-  const { location, url } = sanitizePath(file, addonsDir);
+const os = platform();
 
-  expect(location).toMatchInlineSnapshot(
-    `"/Users/username/Projects/projectname/storybook/node_modules/@storybook/addon-x+y/dist/manager.mjs"`
-  );
-  expect(url).toMatchInlineSnapshot(
-    `"./sb-addons/node_modules/%40storybook/addon-x%2By/dist/manager.mjs"`
-  );
-});
+if (os !== 'win32') {
+  test('sanitizePath', () => {
+    const addonsDir = '/Users/username/Projects/projectname/storybook';
+    const text = 'demo text';
+    const file = {
+      path: '/Users/username/Projects/projectname/storybook/node_modules/@storybook/addon-x+y/dist/manager.mjs',
+      contents: Uint8Array.from(Array.from(text).map((letter) => letter.charCodeAt(0))),
+      text,
+    };
+    const { location, url } = sanitizePath(file, addonsDir);
+
+    expect(location).toMatchInlineSnapshot(
+      `"/Users/username/Projects/projectname/storybook/node_modules/@storybook/addon-x+y/dist/manager.mjs"`
+    );
+    expect(url).toMatchInlineSnapshot(
+      `"./sb-addons/node_modules/%40storybook/addon-x%2By/dist/manager.mjs"`
+    );
+  });
+}
+
+if (os === 'win32') {
+  test('sanitizePath - 1', () => {
```

Comment:
I wonder if it would be better to switch inside the test, e.g. 

```js
    const addonsDir = os === 'win32' ? 'C:\\Users\\username\\Projects\\projectname\\storybook' : '/Users/username/Projects/projectname/storybook';
```

Reply:
Or even a pattern like:

```js
function maybeWindows(path) {
  if (os !=== 'win32') return path;

  return path.startsWith('/') ? `C:\\${slash(path)}` : slash(path);
}

const addonsDir = maybeWindows('/Users/username/Projects/projectname/storybook')
```

Reply:
It's not just the addons dir, which changes, but also the path in line 31 and the assertion in line 37. I would concentrate the windows specific changes into an extra test rather inlining incoherent in three different places. Though, I don't have a strong opinion about that :)

Reply:
I guess I'm just thinking the tests could drift apart if we don't do it inline.

Reply:
Alright! I have inlined the condition.